frontend/src/pages/UserProfile.js

```diff
@@ -0,0 +1,65 @@
+import '../App.css';
+import axios from "axios";
+import React, { useState, useEffect } from 'react';
+import { useNavigate, Link } from 'react-router-dom';
+
+import EditUser from '../components/users/editUser';
+import LogoutUser from '../components/users/logoutUser';
+import DeleteUser from '../components/users/deleteUser';
+
+function UserProfile() {
+    const [user, setUser] = useState({});
+    const [isLoading, setIsLoading] = useState(true); // Add loading state
+    const navigate = useNavigate();
+    
+    useEffect(() => {
+        const loggedInUser = localStorage.getItem('user'); 
+        const getUser = async () => {
+            const user = JSON.parse(loggedInUser);
+            console.log(user);
+            const email = user.email;
+            //find user by stored email
+            axios.post("/user/findByEmail", { email })
+                .then((response) => {
+                    const userObject = response.data.user; // Access the user object from the response
+                    setUser(userObject);
+                    setIsLoading(false); // Set loading state to false after data is fetched
+                })
+                .catch((error) => {
+                    console.error('Error fetching user:', error);
+                    setIsLoading(false); // Set loading state to false in case of an error
+                }
+            );
+        }
+        if (loggedInUser) {
+            getUser();
+        } else {
+            //if no user session
+            navigate('/login');
+        }
+    }, [navigate, setUser]);
```

Comment:
check if can replace `[navigate, setUser]` with `[]`

Reply:
left as [navigate] per warnings given when set to []